### ISSUE TITLE
[QUESTION] Asyncio.sleep seems to block the bot from Processing updates

### ISSUE THREAD

**Issue by Maxwellads** [python-telegram-bot/python-telegram-bot#4423]
### Issue I am facing

So I am working on some code, that decides "delete or not" of previous media group, depending on if an new message is present in channel. After sending the media group, it is natural to wait for a few seconds before the new message could have been sent, right? I tried `asyncio.sleep(10)` for the delay. The update is received by the bot, but it would not be proceeded until the sleep ends. 
```The function is like this
async def delete_msg(update: Update, context: CallbackContext):
    delete_tasks = []
    if caption_succ != 1:
        for cur_id in range(start_msg_id, latest_msg_id + 1, 1):
            delete_tasks.append(context.bot.delete_message(chat_id=channel_id, message_id=cur_id))
        await asyncio.gather(*delete_tasks)
```
If I call the function, it would wait for 10 seconds. During that, updates are received but never proceeded. Maybe the asyncio future of proceeding updates is "delayed" by the sleep? 
Anyways, I am looking for a bypassing solution. Running a scheduler should somehow work, but not in my favor. Just want it to throw a delay of few seconds and being able to process the updates :/

Any help is appreciated. 

### Related part of your code

```python
async def delete_msg(update: Update, context: CallbackContext):
    delete_tasks = []
    if caption_succ != 1:
        for cur_id in range(start_msg_id, latest_msg_id + 1, 1):
            delete_tasks.append(context.bot.delete_message(chat_id=channel_id, message_id=cur_id))
        await asyncio.gather(*delete_tasks)
    
async def handle_messages(update: Update, context: CallbackContext):
    global latest_mediagroup_id, start_msg_id, latest_msg_id, caption_succ, mediagroup_sending
    logging.info("Message Handler Called!!!!!!!!")
    if update.channel_post:
        global channel_id
        channel_id = update.channel_post.chat.id
        if update.channel_post.media_group_id is None:
            logging.info("Received a message while mediagroup is not received. Skipped.")
            await caption_mark()
            return
        else:
            if update.channel_post.media_group_id != latest_mediagroup_id:
                latest_mediagroup_id = update.channel_post.media_group_id
                start_msg_id = update.channel_post.message_id
                logging.debug("New media group: {latest_mediagroup_id}, starting from message: {start_msg_id}")
            else:
                latest_msg_id = update.channel_post.message_id
                logging.info(f"Latest mediagroup id of {channel_id}: {start_msg_id}~{latest_msg_id}")
        if start_msg_id == latest_msg_id - 7:
            await delete_msg(update, context)

        return latest_msg_id
```

### Operating System

All

### Version of Python, python-telegram-bot & dependencies

```shell
python-telegram-bot 21.4
Bot API 7.7
Python 3.12.4 (tags/v3.12.4:8e8a4ba, Jun  6 2024, 19:30:16) [MSC v.1940 64 bit (AMD64)]
```

**Comment by Bibo-Joshi** [comment]
Hi. I have two comments on your code snippets:

* Yes, `await asyncio.sleep` has an impact on handling updates. Please have a look at [this wiki page](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Concurrency)
* You should avoid using globals. In particular, you'll get problems if your bot is handling updates from multiple chats. Plesae have a look at PTBs [built in solution](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Storing-bot%2C-user-and-chat-related-data) for that

**Comment by Maxwellads** [comment]
Thanks for the suggestions. It seems like I have resolved it by passing a large timeout to application, but I will still look onto it as sometimes the connection gets closed instead of timed-out. If the wait pattern cannot be applied, guess I should just leave the deletion task to another bot.
And yes, I am well aware of the global var is some lazy work. I wrote some similar stuff that saves channel infomation into a jsonified file/list and they were not there because I find it troublesome, before I can figure out how to make the feature go as expected. 

Thank you for your dedicated support as always!
